app/test/consistent_apphash_test.go

```diff
@@ -3,134 +3,303 @@ package app_test
+
 	// Create deterministic keys
 	kr, pubKeys := deterministicKeyRing(enc.Codec)
 
-	recs, err := kr.List()
+	// Apply genesis state to the app.
+	valKeyRing, _, err := testutil.SetupDeterministicGenesisState(testApp, pubKeys, 20_000_000_000, app.DefaultInitialConsensusParams())
 	require.NoError(t, err)
-	accountNames := make([]string, 0, len(recs))
 
-	// Get the name of the records
-	for _, rec := range recs {
-		accountNames = append(accountNames, rec.Name)
-	}
+	// ------------ Genesis User Accounts ------------
 
-	// Apply genesis state to the app.
-	_, _, err = testutil.SetupDeterministicGenesisState(testApp, pubKeys, 1_000_000_000, app.DefaultInitialConsensusParams())
-	require.NoError(t, err)
+	// Get account names and addresses from the keyring
+	accountNames := testfactory.GetAccountNames(kr)
+	accountAddresses := testfactory.GetAddresses(kr)
 
 	// Query keyring account infos
 	accountInfos := queryAccountInfo(testApp, accountNames, kr)
 
 	// Create accounts for the signer
-	accounts := make([]*user.Account, 0, len(accountInfos))
-	for i, accountInfo := range accountInfos {
-		account := user.NewAccount(accountNames[i], accountInfo.AccountNum, accountInfo.Sequence)
-		accounts = append(accounts, account)
-	}
+	accounts := createAccounts(accountInfos, accountNames)
 
-	// Create a signer with keyring accounts
+	// Create a signer with accounts
 	signer, err := user.NewSigner(kr, enc.TxConfig, testutil.ChainID, app.DefaultInitialVersion, accounts...)
 	require.NoError(t, err)
 
-	amount := sdk.NewCoins(sdk.NewCoin(app.BondDenom, sdk.NewIntFromUint64(1000)))
+	// ------------ Genesis Validator Accounts  ------------
 
-	// Create an SDK Tx
-	sdkTx := SdkTx{
-		sdkMsgs: []sdk.Msg{
-			banktypes.NewMsgSend(signer.Account(accountNames[0]).Address(),
-				signer.Account(accountNames[1]).Address(),
-				amount),
-		},
-		txOptions: blobfactory.DefaultTxOpts(),
+	// Validators from genesis state
+	genValidators := testApp.StakingKeeper.GetAllValidators(testApp.NewContext(false, tmproto.Header{}))
+
+	// Get validator account names from the validator keyring
+	valAccountNames := testfactory.GetAccountNames(valKeyRing)
+
+	// Query validator account infos
+	valAccountInfos := queryAccountInfo(testApp, valAccountNames, valKeyRing)
+
+	// Create accounts for the validators' signer
+	valAccounts := createAccounts(valAccountInfos, valAccountNames)
+
+	// Create a signer with validator accounts
+	valSigner, err := user.NewSigner(valKeyRing, enc.TxConfig, testutil.ChainID, app.DefaultInitialVersion, valAccounts...)
+	require.NoError(t, err)
+
+	// ----------- Create SDK Messages ------------
+
+	amount := sdk.NewCoins(sdk.NewCoin(app.BondDenom, sdk.NewIntFromUint64(1_000)))
+	// Minimum deposit required for a gov proposal to become active
+	depositAmount := sdk.NewCoins(sdk.NewCoin(app.BondDenom, sdk.NewIntFromUint64(10000000000)))
+	twoInt := sdk.NewInt(2)
+
+	// ---------------- First Block ------------
+	var firstBlockSdkMsgs []sdk.Msg
```

Comment:
[not blocking]
how much harder / less efficient would it be to use table driven tests? This feels like reading the bible

Reply:
I had it in similar format before i changed it to this because i didn't think a super long list of messages was more readable https://github.com/celestiaorg/celestia-app/blob/c01c6b2402586d1919038c9cec28e8f29376b403/app/test/consistent_apphash_test.go#L109

Reply:
but if you think it is i'm happy to change it back to the similar format